src/main/java/com/rabbitmq/http/client/HttpLayer.java

```diff
@@ -0,0 +1,164 @@
+/*
+ * Copyright 2018-2020 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.rabbitmq.http.client;
+
+import com.rabbitmq.http.client.JdkHttpClientHttpLayer.Factory;
+import java.net.URI;
+import java.net.http.HttpClient.Builder;
+import java.net.http.HttpRequest;
+import java.util.Map;
+import java.util.function.Consumer;
+import javax.net.ssl.SSLContext;
+import javax.net.ssl.SSLParameters;
+
+/**
+ * HTTP layer abstraction for {@link Client}.
+ *
+ * <p>This is considered a SPI and is susceptible to change at any time.
+ *
+ * <p>Use the {@link #configure()} to configure and create the default implementation.
+ *
+ * @since 4.0.0
+ */
+public interface HttpLayer {
+
+  /**
+   * Configure the default {@link HttpLayer} implementation.
+   *
+   * @return a configuration instance
+   */
+  static Configuration configure() {
```

Comment:
I would have made the Client constructor to instantiate a JDKHttpLayerFactory rather than adding this method (although static) to the interface.

Reply:
This configuration method is for users that would like to configure the `JdkHttpClientHttpLayer` instance in an easy. `JdkHttpClientHttpLayer` and its `Factory` are package private. Do you think we should make them public instead of using this configuration mechanism?